### PR TITLE
Added segment field

### PR DESCRIPTION
**Pull request by assertchris**
screen shot 2015-08-12 at 9 00 44 am
screen shot 2015-08-12 at 9 00 39 am

### CODE REVIEW

**Comment by tractorcow** [comment]
Can you please fix these test errors?

File: code/model/editableformfields/EditableFormField.php [hunk 1]
```diff
@@ -110,6 +118,13 @@ private function isReadonly() {
 	}
 
 	/**
+	 * @return string
+	 */
+	public function getDisplayName() {
```
Comment (tractorcow) [inline code/model/editableformfields/EditableFormField.php]: Is there a reason to have this separate field anymore?

Comment (assertchris) [inline code/model/editableformfields/EditableFormField.php]: yes. if we require ids/names that differ from the `Name` field (as we did earlier today).

**Comment by tractorcow** [comment]
@assertchris I've added fixes for frontend logic and HTML id generation at https://github.com/open-sausages/silverstripe-userforms/pull/16

It just uses the $field->Name property, if that's what you want to go with. :)

**Comment by tractorcow** [comment]
The SS Templates don't seem to be necessary; With all changes (but those custom templates deleted) I see exactly what was expected.

image

image

**Comment by tractorcow** [comment]
I've deployed this to http://opensausage.test.silverstripe.com/user-form (sans templates) for testing.

It looks like duplicate ID's arent an issue, but I'll spend some time testing the custom rules to make sure.

**Pull request closed without merging**